sdk/core/azure_core/benches/core_benchmarks.rs

```diff
@@ -0,0 +1,31 @@
+use azure_core::http::Url;
+use criterion::{criterion_group, criterion_main, BenchmarkId, Criterion};
+use std::time::Duration;
+// how many params to add to the url
+const PARAMS: &[usize] = &[1, 10, 100, 1000];
+
+fn url_parsing_benchmark(c: &mut Criterion) {
+    let mut group = c.benchmark_group("url_parsing");
+    for num in PARAMS {
+        let mut url = String::new();
+        url.push_str("https://example.com/path/resource?param=value");
+
+        for i in 0..*num {
+            url.push_str(&format!("&param{}=value{}", i, i));
+        }
+        group.bench_function(BenchmarkId::new("parse_basic_url", num), |b| {
+            b.iter(|| {
+                let _url = Url::parse(&url).unwrap();
```

Comment:
1. Best to put this in a `black_box`: <https://github.com/heaths/ordinal-rs/blob/e946015d49e233002212c4ae12022b01642ab6fc/benches/ordinal.rs#L28-L31>
2. Tip: since you don't need the return, just `let _ = ...`.

Reply:
done